salt/grains/core.py

```diff
@@ -494,6 +561,31 @@ def _sunos_memdata():
     return grains
 
 
+def _aix_memdata():
+    '''
+    Return the memory information for AIX systems
+    '''
+    grains = {'mem_total': 0, 'swap_total': 0}
+
+    prtconf = '/usr/sbin/prtconf 2>/dev/null'
+    for line in __salt__['cmd.run'](prtconf, python_shell=True).splitlines():
+        comps = line.split(' ')
+        if comps[0].strip() == 'Memory' and comps[1].strip() == 'Size:':
+            grains['mem_total'] = int(comps[2].strip())
+            break
+
+    swap_cmd = salt.utils.path.which('swap')
+    swap_data = __salt__['cmd.run']('{0} -s'.format(swap_cmd)).split()  # 4k blocks
+    try:
+        swap_avail = int(swap_data[-2]) * 4
+        swap_used = int(swap_data[-6]) * 4
+        swap_total = (swap_avail + swap_used)
```

Comment:
Why not just:
```python
swap_total = int(swap_data[-2]) * 4 + int(swap_data[-6]) * 4
```

Reply:
And I am not sure if this is the entirely right way, but my AIX knowledge greatly outdated. Back in times I was using `lsps` instead, since it shows disks that are swap allocated, but some of them might be inactive at this particular moment. Maybe you would like consider to take this to the account? Correct me if I am wrong.

Reply:
code changed
prefer to use similar commands across the various UNIX's as much as possible, Solaris is using swap.